chia-client/src/peer.rs

```diff
@@ -106,10 +70,144 @@ impl Peer {
                 (3, "1".to_string()),
             ],
         };
-        self.send(handshake).await
+        self.send(body).await
+    }
+
+    pub async fn request_puzzle_and_solution(
+        &self,
+        coin_id: Bytes32,
+        height: u32,
+    ) -> Result<PuzzleSolutionResponse, RejectPuzzleSolution> {
+        let body = RequestPuzzleSolution {
+            coin_name: coin_id,
+            height,
+        };
+        let response: RespondPuzzleSolution = self.request(body).await?;
+        Ok(response.response)
+    }
+
+    pub async fn send_transaction(&self, spend_bundle: SpendBundle) -> Result<TransactionAck, ()> {
+        let body = SendTransaction {
+            transaction: spend_bundle,
+        };
+        self.request_infallible(body).await
+    }
+
+    pub async fn request_block_header(
+        &self,
+        height: u32,
+    ) -> Result<HeaderBlock, RejectHeaderRequest> {
+        let body = RequestBlockHeader { height };
+        let response: RespondBlockHeader = self.request(body).await?;
+        Ok(response.header_block)
+    }
+
+    pub async fn request_block_headers(
+        &self,
+        start_height: u32,
+        end_height: u32,
+        return_filter: bool,
+    ) -> Result<Vec<HeaderBlock>, ()> {
+        let body = RequestBlockHeaders {
+            start_height,
+            end_height,
+            return_filter,
+        };
+        let response: RespondBlockHeaders =
+            self.request(body)
+                .await
+                .map_err(|error: Error<RejectBlockHeaders>| match error {
+                    Error::Rejection(_rejection) => Error::Rejection(()),
+                    Error::Chia(error) => Error::Chia(error),
+                    Error::WebSocket(error) => Error::WebSocket(error),
+                    Error::InvalidResponse(error) => Error::InvalidResponse(error),
+                    Error::MissingResponse => Error::MissingResponse,
+                })?;
```

Comment:
I'm a bit confused by this. what happens to these error codes? the function is declared to return `()` as the error.

Reply:
I think it may be confusing to read since `Result<ResponseMessage, RejectionMessage>` is actually a custom type, shadowing the normal `std::result::Result<T, ErrorType>`. The `()` just means than the error variant is `Error<()>`, meaning no rejection message but still other types of errors can be returned.

Would it be better if I rename `Result` to `Response` or something like that?

Reply:
oh, the `Result<>` type here is not `std::result::Result`? I guess that's where my confusion came from.

Reply:
Fixed